### PR TITLE
Use own linux runner for lint

### PR DESCRIPTION
**Pull request by zhouzhuojie**
Seeing many issues from the lint.yml that I'm guessing if we run lint on our own runners, things may improve.

- https://github.com/pytorch/pytorch/pull/59760/checks?check_run_id=2788684303
- https://github.com/pytorch/pytorch/pull/59760/checks?check_run_id=2788683792

### CODE REVIEW

**Comment by samestep** [comment]
aw but this is so sad 😢 now lint is no longer free

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ on:
 
 jobs:
   quick-checks:
-    runs-on: ubuntu-18.04
+    runs-on: linux.2xlarge
     steps:
       - name: Setup Python
         uses: actions/setup-python@v2
```
Comment (walterddr) [inline .github/workflows/lint.yml:14]: need to change `python-version` seems like 3.x is not recognized on our linux.2xlarge

also do we have something smaller than c5.2xlarge for linter, if our goal is stability not TTS, dont think linter requires that much firepower

Comment (walterddr) [reply .github/workflows/lint.yml:14]: also according to [this doc](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources), free tier ubuntu-18.04 is c5.large?

**Review by walterddr — APPROVED** [review]
let's get this in first - given the many recent issues with gha linter issues (plz fix python version)

**Comment by janeyx99** [comment]
Is the reason behind this improvement the fact that we have more control over our runners compared to using GitHub's machines which we have no control over?

**Comment by zhouzhuojie** [comment]
> Is the reason behind this improvement the fact that we have more control over our runners compared to using GitHub's machines which we have no control over?

there're limits on the free runner https://docs.github.com/en/actions/reference/usage-limits-billing-and-administration#usage-limits, I'm guessing it's related to the 20 concurrent jobs, but need more data to verify

File: .github/workflows/lint.yml (hunk 1 above)
Comment (zhouzhuojie) [reply .github/workflows/lint.yml:14]: @seemethere do you know where can I find the definition of the image or ami used for linux.2xlarge? wondering how isolated is it and how to setup correct python environment for it

**Comment by seemethere** [comment]
Should we create a smaller instance type specifically for these use cases?

Seems like a `linux.2xlarge` might be a little overpowered for these

**Pull request closed without merging**